### ISSUE TITLE
Potential bad package added in AndroidTools.java

### ISSUE THREAD

**Issue by Tnnnguyen** [smartdevicelink/sdl_android#691]
### Bug Report
We are adding the package info to the list even when the system cannot find the package and throws a NameNotFoundException. This can lead to a bad ResolveInfo and PackageInfo being used to create a SdlAppInfo object and subsequently a library crash. [Reference](https://github.com/smartdevicelink/sdl_android/blob/develop/sdl_android/src/main/java/com/smartdevicelink/util/AndroidTools.java#L104-L112).

```
for (ResolveInfo info : resolveInfoList) {
					PackageInfo packageInfo = null;
					try {
						packageInfo = packageManager.getPackageInfo(info.serviceInfo.packageName, 0);
					} catch (NameNotFoundException e) {
					} finally {
						sdlAppInfoList.add(new SdlAppInfo(info, packageInfo));
					}
				}
```

The code above adds every package to the list regardless because it's being done inside the finally clause. Should it pass on the package that results in an exception?

**Comment by joeygrover** [comment]
The `ResolveInfo` should theoretically be fine, and we only use the `PackageInfo` for the timestamps of last updated and first installed; both of which are not critical pieces of info for the creation of the `SdlAppInfo`. However, I agree that if the package manager can't find the package it is likely that the supplied `ResolveInfo` may have incorrect/corrupt data so we could assume it is of no use and should be discarded.

**Comment by joeygrover** [comment]
Fixed with #692
